### PR TITLE
Deprecate the use of `entities.System` in favor of `entities.Host`

### PR DESCRIPTION
**Pull request by abalakh**
Part of #270

### CODE REVIEW

**Comment by abalakh** [comment]
`[DONOTMERGE]` tag as i'd like to add some unit tests not to reduce coverage. Will work on them when receive all the comments/acks regarding this PR.

File: nailgun/entities.py [hunk 1]
```diff
@@ -1888,6 +1890,18 @@ def create_payload(self):
...
         return payload
 
+    def read(self, entity=None, attrs=None, ignore=None):
+        """Ignore 'host' field as it is not returned by the server.
+
+        For more information, see `Bugzilla #1325989
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1325989>`_.
+
+        """
+        if ignore is None:
```
Comment (elyezer) [inline nailgun/entities.py]: Would not be required to copy the host attr?

File: nailgun/entities.py [hunk 2]
```diff
@@ -2000,6 +2014,53 @@ def update_payload(self, fields=None):
         return {u'hostgroup': super(HostGroup, self).update_payload(fields)}
 
 
+class HostPackage(Entity):
+    """A representation of a Host Package entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        if _get_version(server_config) < Version('6.2'):
+            raise NotImplementedError(
```
Comment (elyezer) [inline nailgun/entities.py]: I saw this and it makes me wonder if would be good to add a deprecation warning when the user access the system entity on Satellite >= 6.2?

**Comment by elyezer** [comment]
The changes looks good to me. ACK

File: nailgun/entities.py (hunk 2 above)
Comment (abalakh) [inline nailgun/entities.py]: You mean [this](https://github.com/SatelliteQE/nailgun/pull/276/files/0f714442b656ac44bbb5da476bd8d407b35a0df1#diff-3822710a997c3f2f7c3e93e884af69caR4183) warning? :)

Comment (elyezer) [inline nailgun/entities.py]: That one! I just did not see that! Good job

File: nailgun/entities.py (hunk 1 above)
Comment (abalakh) [inline nailgun/entities.py]: I don't need host attrs here, i'm just passing received attrs to super with no changes. Purpose of this `read()` is to add 1 field to ignore and that's it. Anyway, that's a temporary solution - mentioned BZ should be fixed pretty soon and whole `read()` will be dropped.

**Comment by abalakh** [comment]
OK, covered 12 of 13 missed lines. The only thing that wasn't covered - `HostCollection.read()`. I skipped it on purpose - that `read()` is temporary workaround which will be dropped really soon.

**Comment by abalakh** [comment]
Hm... On the other hand, it takes no extra efforts to cover it and drop corresponding test lines together with `HostCollection.read()` when time comes. Will update the PR in a moment

**Comment by elyezer** [comment]
@abalakh :+1:

**Comment by oshtaier** [comment]
So what was final decision about `System` and `cvs`?
```
system = entities.System(
    content_view=content_view,
    environment=lc_env,
    organization=org,
).create()
```

**Comment by abalakh** [comment]
No final decision here, that's why i put a comment that this PR is a `Part of #270`, not `Closes #270` :) Planned to work on that right after this PR, but i just took a brief look as you reminded and it seems as a simple API bug about missing `content_view_id` attribute for me: we had this ability for system path, we have it via WebUI for hosts too, but for some reason content views are not accepted via API for hosts. So I'll submit a bug and we'll see how it's going.

**Comment by abalakh** [comment]
To make it more clear:
To completely switch from System entity to Host entity we need to:
Nailgun:
* This PR, which deprecates system, adds host attribute where it's needed, etc, etc,
* Un-ignore host attribute for HostCollection entity as soon as BZ1325989 is fixed
* Add content_view field to Host entity as soon as it's added to Satellite

Robottelo:
* Fix HostCollection tests (already done, corresponding PR is in place)
* Fix 2 tests with System entity and content views specified as soon as BZ### is fixed
* Fix robottelo's multiple_path tests
* Fix API_PATHS in test_api_smoke.py

Basically, the biggest ones are the first for nailgun and robottelo, which are already done, all the rest are simple 1-3 lines fixes which are very easy to do, but we need corresponding bugs to be closed first.

**Comment by oshtaier** [comment]
ACK
